nativelib/src/main/scala/scala/scalanative/unsafe/Zone.scala

```diff
@@ -54,6 +54,7 @@ object Zone {
       if (rawptr == null) {
         throw new OutOfMemoryError(s"Unable to allocate $size bytes")
       }
+      libc.memset(rawptr, 0, size)
```

Comment:
You are absolutely correct that having package.scala alloc and z.alloc have the same name but different behaviors when it comes to clearing is a nightmare. I have been 
hosed by that more times that I care to admit in public, or even in to myself in the dead and quiet of night.

I believe the discussion of memory handling here is larger than proper for a review comment.  I will create an Issue to describe my concerns.  In short, I think that
one needs to be able to create either raw or cleared memory.   

See line 199 for the case which concerns me, why allocate cleared memory
when one _knows_ that it will be immediately overwritten?

Reply:
Limiting myself to only the suggested change. From my reading, in most c runtime libraries, especially libc, he combination `malloc/memset` is not directly equivalent to
`calloc` and the latter is preferred.  At the very least, `calloc` does a check for
integer overflow (yes, that slows it down by a few cycles).  I understand that 
`calloc` for more than 128 KiB can use some memory management techniques to delay the zeroing and otherwise improve performance.

A small improvement, however, a well know one and, IMHO, well worth reaping.
I suspect/believe that you went for a minimal change for existing code.

Reply:
@LeeTibbert I've spent a lot of time before I realised this *small* different. And I strongly suggest to apply it because to find this the second time is a nightmare.

Reply:
Yes,  to methods with the same name and opposite behavior is at best an H. P. Lovecraft  nightmare 
and probably worse.  

I agree with fixing that and propose that the alloctor of cleared memory  use `calloc` instead
of the `memset/memcpy` of this region. This is for all the reasons that you have described
in your other PR.

This OK as is good and, in my opinion should advance. I believe that it would be better
with the suggested `calloc` change, if only to calloc(1, size).

It is not a direct concern of this PR, but I believe that it needs a __prominent__ release
note. This change increases the likelihood  of correct use, but also has an impact on
performance. Devos should be made aware of that.
----------------------------------------------------------------------------------------------------------------
My concern is that there should be a raw allocator, as well as a cleared allocator.  

There are roughly 141 instances of " = alloc" now in the SN project.  To advance things, in
another PR and after discussion I could create an `allocRaw()` and  look at each of those
instances and see if it should clear or not. That would be a companion PR and limit
the performance impact of this PR.

Reply:
Did I mention that your fixing this is a Mercy?  Yes, it is, thank you.

Reply:
@LeeTibbert technically `calloc` is the same `malloc` and `memset` at least it is true for GNU libc.

Anyway, I see one differences between `calloc` and `malloc` + `memset` and it is a way that allow to optimize `calloc` to allocate `zeros pages` from the kernel and allow it to make overcommit for memory for SN application.

I'm thinking about this use case and the more I think the more I agree with idea that it is bad because it open very bad door for SN user that allows to allocate more memory than system has and when applicatio treis to use it, it will be killed by OOM killer.

Basicly this "unsure" is a reason why I've created this trivial PR to fix this issue, and move `calloc` vs `malloc` + `memset` to future release.